api/prometheus/v1/api.go

```diff
@@ -1138,6 +1138,9 @@ func (h *apiClientImpl) DoGetFallback(ctx context.Context, u *url.URL, args url.
 		return nil, nil, nil, err
 	}
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	// Underlying `net.http` library automatically retries` idempotent requests when connectivity issues are hit.
+	// POST requests are not considered idempotent by default, so we need to explicitly mark them as such.
+	req.Header.Set("Idempotency-Key", "TODO")
```

Comment:
Not sure which value we should set. It doesn't really matter as the go implementation only checks for the [header presence](https://github.com/golang/go/blob/eca0d44cec58951fb716e540dcc21c0f527686d5/src/net/http/request.go#L1429).

On the other hand this [IEFT draft](https://tools.ietf.org/id/draft-idempotency-header-01.html#name-uniqueness-of-idempotency-k) document claims that it must be unique and it is recommended to use UUID or similar function.

I made some research and this project doesn't use any UUIC library. [`mysql_exporter`](https://github.com/prometheus/mysqld_exporter/blob/main/go.mod#L7) utilizes `github.com/google/uuid` so that might be a good candidate. Another option would be to utilize [xxhash](https://github.com/prometheus/client_golang/blob/main/go.mod#L5) module, which is already utilized in `client_golang`.

Reply:
I would suggest `req.Header["Idempotency-Key"] = nil`, given the [docs ](https://pkg.go.dev/net/http#Transport) talking about "value is a zero-length slice"

Reply:
I missed that part. Thanks, updated my PR.